.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,127 @@
+name: CI/CD
+
+on:
+  push:
+    branches:
+      - nu
+  pull_request:
+
+jobs:
+  knossos:
+    name: Knossos / ${{ matrix.archive }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-2019, macos-10.15, ubuntu-20.04]
+        include:
+          - os: windows-2019
+            script: .\task.ps1
+            archive: windows-x64
+          - os: macos-10.15
+            script: ./task.sh
+            archive: darwin-x64
+          - os: ubuntu-20.04
+            script: ./task.sh
+            archive: ubuntu-x64
+      fail-fast: false
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Reading additional settings from repository
+        id: vars
+        shell: bash
+        run: |
+          msys2_extras=$(grep -o 'pacman --noconfirm -Syu --needed [^"]*' tasks.star | cut -d ' ' -f 5-)
+          msys2_extras=$(sh -c "echo $msys2_extras") # subshell echo to perform brace expansion
+          echo "Adding ${msys2_extras} to MSYS2"
+          echo "::set-output name=msys2_extras::${msys2_extras}"
+          go_version=$(cat .go-version)
+          echo "Using Go version ${go_version}"
+          echo "::set-output name=go_version::${go_version}"
+
+      - name: Setup MSYS2 toolchain (Windows)
+        uses: msys2/setup-msys2@v2
+        if: runner.os == 'Windows'
+        with:
+          update: true
+          install: ${{ steps.vars.outputs.msys2_extras }}
+
+      - name: Setup Go toolchain
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ steps.vars.outputs.go_version }}
+
+      - name: Cache Go dependencies
+        uses: actions/cache@v2
+        id: golang-cache
+        with:
+          key: golang-${{ runner.os }}-${{ hashFiles('**/go.sum') }}
+          restore-keys: golang-${{ runner.os }}-
+          path: ~/go/pkg/mod
+
+      - name: Cache Yarn dependencies
+        uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          key: yarn-${{ runner.os }}-${{ hashFiles('yarn.lock') }}
+          restore-keys: yarn-${{ runner.os }}-
+          path: .yarn/cache
+      
```

Comment:
~~We have to cache `.yarn/unplugged` as well because packages like `node-sass` store downloaded files there.~~

Now it caches `.yarn/releases` and `.yarn/plugins` as well which will probably cause issues after the next Yarn update since the cached (outdated) version will replace the newer repo version, wouldn't it?

Reply:
Hmm, not sure it matters as on a new yarn release our `yarn.lock` will change as well, invalidating the cache. Still, wouldn't hurt to ignore it? I just liked the more minimal pattern.

Reply:
I don't think yarn or plugin updates actually update the `yarn.lock` file. Yarn updates would update the `.yarnrc.yml` since the filename is versioned but plugin updates only update the `.yarn/plugins/...cjs` file which is a bit annoying since it's almost impossible to check for updates.

Reply:
Added the ignores!